### PR TITLE
Add an inbox getAll endpoint.

### PR DESCRIPTION
**Pull request by mattcollier**
Dependent PR: https://github.com/digitalbazaar/bedrock-ldn-inbox/pull/2

See note on that PR wrt possible DOS style attack.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -155,13 +157,45 @@ bedrock.events.on('bedrock-express.configure.routes', app => {
...
+        }
+        let query = {};
+        let fields = {};
+        let options = {};
+        if(req.query.owner) {
+          query.owner = database.hash(req.query.owner);
+        } else {
+          query.owner = database.hash(identity.id);
+        }
+        brLdnInbox.inboxes.getAll(identity, query, fields, options, callback);
```
Comment (dlongley) [inline lib/index.js]: This should not return records directly but rather just inboxes.

File: lib/index.js [hunk 2]
```diff
@@ -155,13 +157,45 @@ bedrock.events.on('bedrock-express.configure.routes', app => {
         }).type('application/ld+json').send(inbox));
   });
 
+  // get inbox info, including list of messages
```
Comment (dlongley) [inline lib/index.js]: Is this comment right? Are lists of messages returned? If so, is that what we want? I would think not.

**Review by dlongley — CHANGES_REQUESTED** [review]

**Comment by mattcollier** [comment]
@dlongley ready for another pass.

File: lib/index.js [hunk 3]
```diff
@@ -155,13 +157,47 @@ bedrock.events.on('bedrock-express.configure.routes', app => {
...
+        if(req.user && req.user.identity) {
+          identity = req.user.identity;
+        }
+        let query = {};
+        let fields = {};
+        let options = {};
+        if(req.query.owner) {
+          query.owner = database.hash(req.query.owner);
+        } else {
+          query.owner = database.hash(identity.id);
```
Comment (dlongley) [inline lib/index.js]: Do we actually want this as the default? I think there should be no default.

Comment (mattcollier) [reply lib/index.js]: removed.

**Comment by mattcollier** [comment]
@dlongley ready for another pass.

**Review by dlongley — APPROVED** [review]

**Pull request merged**